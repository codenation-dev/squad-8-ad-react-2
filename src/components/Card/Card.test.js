import React from "react";
import renderer from "react-test-renderer";
import "jest-styled-components";

import {
  CardBox,
  CardTitle,
  CardDescription,
  Details,
  DetailsContainer
} from "./styles";

test("CardBox renders correctly", () => {
  const tree = renderer.create(<CardBox />).toJSON();
  expect(tree).toMatchSnapshot();
});

test("CardTitle renders correctly", () => {
  const tree = renderer.create(<CardTitle />).toJSON();
  expect(tree).toMatchSnapshot();
});

test("CardDescription renders correctly", () => {
  const tree = renderer.create(<CardDescription />).toJSON();
  expect(tree).toMatchSnapshot();
});

test("Details renders correctly", () => {
  const tree = renderer.create(<Details />).toJSON();
  expect(tree).toMatchSnapshot();
});

test("DetailsContainer renders correctly", () => {
  const tree = renderer.create(<DetailsContainer />).toJSON();
  expect(tree).toMatchSnapshot();
});
