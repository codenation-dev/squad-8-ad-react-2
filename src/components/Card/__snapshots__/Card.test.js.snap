// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardBox renders correctly 1`] = `
.c0 {
  width: 350px;
  min-height: 125px;
  word-wrap: break-word;
  margin-top: 2px;
  box-shadow: 0 1px 3px rgba(0,0,0,0.12),0 1px 2px rgba(0,0,0,0.24);
  border-radius: 4px;
}

<div
  className="c0"
/>
`;

exports[`CardDescription renders correctly 1`] = `
.c0 {
  padding: 15px;
  margin-top: 10px;
  font-size: 14px;
  color: rgba(0,0,0,0.87);
}

<p
  className="c0"
/>
`;

exports[`CardTitle renders correctly 1`] = `
.c0 {
  padding: 10px 15px;
  margin-bottom: 5px;
  border-bottom: 1px solid #eee;
}

.c0 div {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  justify-items: center;
}

.c0 div span {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  background-color: #085191;
  color: #fff;
  text-transform: uppercase;
  padding: 10px;
  width: 40px;
  height: 40px;
  font-size: 16px;
  border-radius: 50%;
}

.c0 div div {
  display: block;
  margin-left: 8px;
  font-size: 14px;
  font-weight: 400;
  line-height: 1.43;
}

.c0 div div h3 {
  width: 270px;
  white-space: nowrap;
  text-overflow: ellipsis;
  display: block;
  overflow: hidden;
  font-weight: 400;
  color: rgba(0,0,0,0.87);
}

.c0 div div strong {
  font-size: 14px;
  font-weight: 400;
  color: rgba(0,0,0,0.54);
}

<div
  className="c0"
/>
`;

exports[`Details renders correctly 1`] = `
.c0 {
  margin-top: 10px;
  font-size: 14px;
  color: #999;
  font-style: italic;
}

<span
  className="c0"
/>
`;

exports[`DetailsContainer renders correctly 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  padding: 10px;
}

<div
  className="c0"
/>
`;
